src/pass/loop_partition.cc

```diff
@@ -509,46 +509,42 @@ Stmt LoopPartitioner::TryPartition(const Node* node,
   bool pre_stmt_recurse = true;
   if (middle_interval_i->HasLowerBound()) {
     body_begin = ir::Simplify(middle_interval.min());
-    if (!analyzer_.CanProve(body_begin == min)) {
```

Comment:
this check is removed as we are doing the same later on line 514. 

`body_begin - min >= 0` . Let'say `body_begin == min` then, `pre_stmt` will be of zero extent which will later be remove by `RemoveNoOp`. 

Same goes with `post_stmt`.